### PR TITLE
FIX: ssh LocalForward listen only on localhost

### LINKED ISSUE #8: Ensure ssh port forwarding only listens on local interface
* Ensure ssh port forwarding only listens on local interface
* LocalForward can default to "all interfaces on local machine"

### PR DESCRIPTION
**Pull request by klauer**
Closes #8

### CODE REVIEW

**Review by ZLLentz — APPROVED** [review]
I don't know the implications of all of these, but it looks reasonable

**Comment by tangkong** [comment]
Looks good to me too!  I was too slow :<

File: ssh/off_site_config [hunk 1]
```diff
@@ -27,9 +27,9 @@ Host pslogin-arch
     HostName pslogin.slac.stanford.edu
     # Archiver management interface:
     # use http://localhost:17665/mgmt/ui/index.html
-    LocalForward 17665 pscaa02:17665
+    LocalForward localhost:17665 pscaa02:17665
```
Comment (klauer) [inline ssh/off_site_config:30]: Implication:
`LocalForward 17665 ...` may listen on all network interfaces
whereas
`LocalForward localhost:17665 ...` will listen only on your loopback interface

Such that if you were to join for cafe day, you would offer up your port forwarding to everyone else in the coffee shop.

File: ssh/off_site_config [hunk 2]
```diff
@@ -44,18 +44,20 @@ Host psdev
 Host psbuild-plc
     HostName psbuild-rhel7
     ProxyJump pslogin
+    ExitOnForwardFailure yes
```
Comment (klauer) [inline ssh/off_site_config:47]: This host is _just_ for port forwarding; so if it fails, the ssh connection just bails right away

File: ssh/off_site_config [hunk 3]
```diff
@@ -44,18 +44,20 @@ Host psdev
...
-
+    RemoteCommand while true; do echo "** Ready for remote desktop connections (ctrl-c to exit) **"; read -n 1; done
+ 
 Host psbuild-socks
     HostName psbuild-rhel7
     ProxyJump pslogin
     ForwardAgent no
-    DynamicForward 8080
-    RemoteCommand echo "** SOCKS proxy ready **"; read -n 10
+    DynamicForward localhost:8080
```
Comment (klauer) [inline ssh/off_site_config:58]: This sets up the socks proxy on localhost port 8080

File: ssh/off_site_config [hunk 4]
```diff
@@ -44,18 +44,20 @@ Host psdev
...
+    ExitOnForwardFailure yes
+    RemoteCommand while true; do echo "** SOCKS proxy ready (ctrl-c to exit) **"; read -n 1; done
```
Comment (klauer) [inline ssh/off_site_config:60]: This command just runs on the remote host so you don't reuse it for regular ssh purposes

**Pull request merged**